### PR TITLE
[TS] LPD-1849 Avoid closing datasources that were retrieved via JNDI lookup

### PR DESCRIPTION
**Pull request by ericyanLr**
Hi Team,

**Ticket:** https://liferay.atlassian.net/browse/LPD-1849

**Context**
The issue in this ticket was briefly discussed during one of our weekly meetings. 

To summarize: we decided to treat this issue as a bug
- **Reason:** Liferay doesn't own the datasource and so we shouldn't be closing it
- **Suggested Fix:** if incoming datasource is given to us, wrap it with a wrapper

**Proposed Fix**
From my understanding of our code, we prefer using our **DataSourceFactory** classes to manage adding / removing datasources. And these classes utilize our **DataSourceWrapper** class to do additional handling.

And so, the approach I took was to implement a new **JNDIDataSourceWrapper** that would simply be used to identify datasources that are being introduced via JNDI lookup and to avoid these datasources from being closed.

My assumption is that for custom code that involves using an external data source, we would prefer developers to also utilize our **DataSourceFactory** classes to manage these datasources, such as when adding an external data source to a Service Builder module. And so, with this fix, if a customization is utilizing our **DataSourceFactory** classes to retrieve a JNDI data source, our code would no longer be closing this connection unintentionally.

Please let me know if you have any questions.
Thanks!

### CODE REVIEW

**Comment by ericyanLr** [comment]
ci:test:sf

**Comment by ericyanLr** [comment]
ci:test:relevant

**Comment by ericyanLr** [comment]
ci:report:2532248363

**Comment by ericyanLr** [comment]
**ci:report analysis**
Looking at the report, the listed unique failures were also found in previous test results.

Marking this as **Ready to Review (no regression)**

/cc @tinatian

File: portal-impl/src/com/liferay/portal/dao/jdbc/util/JNDIDataSourceWrapper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ */
+
+package com.liferay.portal.dao.jdbc.util;
+
+import javax.sql.DataSource;
+
+/**
+ * @author Eric Yan
+ */
+public class JNDIDataSourceWrapper extends DataSourceWrapper {
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portal/dao/jdbc/util/JNDIDataSourceWrapper.java:13]: @ericyanLr Please make this a private static class of DataSourceFactoryImpl

**Comment by tinatian** [comment]
@ericyanLr Is there any test cover this? If not, please create one.

Thanks.
Tina

**Pull request closed without merging**